### PR TITLE
Bowling-challenge-js

### PR DESCRIPTION
**Pull request by TomasGarciaDev**
To those who will review this challenge, it's almost finished. I still have to check and set up 2 or more scenarios. Thanks in advance!

### CODE REVIEW

File: tests/bowlingGame.test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+const BowlingGame = require("../lib/bowlingGame");
+
+let bowlingGame;
+beforeEach(() => {
```
Comment (j-cort) [inline tests/bowlingGame.test.js:4]: this is a nice way to keep your code DRY.

File: tests/bowlingGame.test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  bowlingGame = new BowlingGame();
+});
+
+describe("bowlingGame", () => {
+  it("can create and instance of bowlingGame", () => {
+    expect(bowlingGame).toBeInstanceOf(BowlingGame);
+  });
+
+  it("can play gutter game", () => {
```
Comment (j-cort) [inline tests/bowlingGame.test.js:13]: Tests are easy to understand!

File: tests/bowlingGame.test.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    bowlingGame.roll(5);
+    bowlingGame.roll(10);
+    bowlingGame.roll(5);
+    bowlingGame.roll(3);
+    for (i = 0; i < 14; i++) {
+      bowlingGame.roll(0);
+    }
+    expect(bowlingGame.getScore()).toBe(46);
+  })
+});
```
Comment (j-cort) [inline tests/bowlingGame.test.js:78]: Maybe we could add some testing of edge cases (e.g. if a user rolls '11' or if a user rolls a non-integer)

File: lib/bowlingGame.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
+class BowlingGame {
```
Comment (j-cort) [inline lib/bowlingGame.js:1]: Maybe naming of variables and functions could be a little clearer (or add an explanation in the readme). It's a little hard to understand the differences between 'strikeCounter' vs 'strikeCounter2' / 'strikeAndstrikeLogic2' vs strikeAndstrikeLogic3' etc.

File: lib/bowlingGame.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  sum(array) {
+    let total = 0;
+    array.map((num) => {
+      return (total += num);
+    });
+    return total;
+  }
+}
+
+module.exports = BowlingGame;
```
Comment (j-cort) [inline lib/bowlingGame.js:115]: I think the code is really nice and concise. 115 lines is quite good for such a complex game.

File: lib/bowlingGame.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+      this.strikeLogicPart1();
+    } else if (this.spareCounter == 10) {
+      this.spareLogicPart2(pins);
+    } else if (this.isSpare()) {
+      this.spareLogicPart1();
+    } else if (this.frame.length === 2) {
+      this.normalFrameLogic();
+    }
+  }
+
```
Comment (j-cort) [inline lib/bowlingGame.js:35]: Looks like an efficient way to calculate the score.

**Review by j-cort — COMMENTED** [review]
Sorry for the delay, I've finished the review now. Great project!
